client/telemetry/TelemetryEventClient.go

```diff
@@ -0,0 +1,297 @@
+package telemetry
+
+import (
+	"encoding/json"
+	"fmt"
+	client "github.com/devtron-labs/devtron/client/events"
+	"github.com/devtron-labs/devtron/client/pubsub"
+	"github.com/devtron-labs/devtron/internal/sql/repository"
+	util2 "github.com/devtron-labs/devtron/internal/util"
+	"github.com/devtron-labs/devtron/pkg/cluster"
+	"github.com/devtron-labs/devtron/pkg/user"
+	"github.com/devtron-labs/devtron/util"
+	"github.com/go-pg/pg"
+	"github.com/posthog/posthog-go"
+	"go.uber.org/zap"
+	"gopkg.in/robfig/cron.v3"
+	"k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type TelemetryEventClientImpl struct {
+	cron                 *cron.Cron
+	logger               *zap.SugaredLogger
+	client               *http.Client
+	clusterService       cluster.ClusterService
+	K8sUtil              *util2.K8sUtil
+	aCDAuthConfig        *user.ACDAuthConfig
+	config               *client.EventClientConfig
+	environmentService   cluster.EnvironmentService
+	userService          user.UserService
+	appListingRepository repository.AppListingRepository
+	PosthogClient        *pubsub.PosthogClient
+}
+
+type TelemetryEventClient interface {
+}
+
+func NewTelemetryEventClientImpl(logger *zap.SugaredLogger, client *http.Client, clusterService cluster.ClusterService,
+	K8sUtil *util2.K8sUtil, aCDAuthConfig *user.ACDAuthConfig, config *client.EventClientConfig,
+	environmentService cluster.EnvironmentService, userService user.UserService,
+	appListingRepository repository.AppListingRepository, PosthogClient *pubsub.PosthogClient) (*TelemetryEventClientImpl, error) {
+	cronLogger := &CronLoggerImpl{logger: logger}
+	cron := cron.New(
+		cron.WithChain(
+			cron.SkipIfStillRunning(cronLogger),
+			cron.Recover(cronLogger)))
+	cron.Start()
+	watcher := &TelemetryEventClientImpl{
+		cron:   cron,
+		logger: logger,
+		client: client, clusterService: clusterService,
+		K8sUtil: K8sUtil, aCDAuthConfig: aCDAuthConfig, config: config,
+		environmentService: environmentService, userService: userService,
+		appListingRepository: appListingRepository, PosthogClient: PosthogClient,
+	}
+	logger.Info()
+	_, err := cron.AddFunc(fmt.Sprintf("@every %dm", 3), watcher.SummeryEventForTelemetry)
+	if err != nil {
+		fmt.Println("error in starting summery event")
+		return nil, err
+	}
+
+	_, err = cron.AddFunc(fmt.Sprintf("@every %dm", 1), watcher.HeartbeatEventForTelemetry)
```

Comment:
if orchestrator bounces timer will get reset

Reply:
in that case we can get more than one event for heartbeat is this will create problem ?

Reply:
solved